scss/_patterns_navigation.scss

```diff
@@ -651,6 +695,158 @@ $spv-navigation-logo-bottom-position: 0.125rem; // 2px when 1rem is 16px
     }
   }
 
+  // SLIDING NAVIGATION
+  .p-navigation.is-sliding {
+    overflow: hidden; // stops overflow scrolling on the page on smaller screens
+
+    @media (min-width: $breakpoint-navigation-threshold) {
+      overflow: visible;
+    }
+
+    // initialise navigation nav as sliding nav on smaller screens
+    .p-navigation__nav {
+      @extend %navigation-slide;
+
+      overflow-x: hidden;
+      transform: translateX(0);
+
+      // reset navigastion nav to default on larger screens
+      @media (min-width: $breakpoint-navigation-threshold) {
+        display: flex;
+        flex-direction: row;
+        height: auto;
+        justify-content: space-between;
+        margin-right: map-get($grid-margin-widths, default);
+        overflow-x: visible;
+        overflow-y: visible;
+        position: static;
+        transform: translateX(0);
+        width: 100%;
+      }
+    }
+
+    &:target,
+    &.has-menu-open {
+      height: 100vh; // make sure navigation take up the whole screen when open on smaller screens
+
+      // reset height to default on larger screens
+      @media (min-width: $breakpoint-navigation-threshold) {
+        height: auto;
+        overflow-x: visible;
+      }
+
+      // slide in navigation nav when navigation is open
+      .p-navigation__nav {
+        transform: translateX(0);
+      }
+    }
+
+    // align arrow icon with text baseline
+    .p-navigation__item--dropdown-toggle .p-navigation__link::after {
+      @media (max-width: calc($breakpoint-navigation-threshold - 1px)) {
+        top: $spv--large;
+      }
+    }
+
+    .p-navigation__item--dropdown-toggle.is-active > .p-navigation__dropdown,
+    .p-navigation__item--dropdown-toggle.is-active > .p-navigation__dropdown--right {
+      // reset dropdown to default display for larger screens
+      @media (min-width: $breakpoint-navigation-threshold) {
+        display: block;
+      }
+    }
+
+    // move arrow icon to the left for back button
+    .p-navigation__item--dropdown-toggle.is-back .p-navigation__link::after {
+      left: map-get($grid-margin-widths, small);
+      transform: rotate(90deg);
+
+      @media (min-width: $threshold-4-6-col) {
+        left: map-get($grid-margin-widths, default);
+      }
+    }
+
+    .p-navigation__item--dropdown-toggle.is-back {
+      @media (min-width: $breakpoint-navigation-threshold) {
+        display: none; // hide back button on larger screens
+      }
+    }
+
+    // initialise navigation dropdown as sliding nav on smaller screens
+    .p-navigation__dropdown,
+    .p-navigation__dropdown--right {
+      @extend %navigation-slide;
+
+      box-shadow: none;
+      display: block;
+      overflow-x: hidden;
+      top: 0;
+
+      // reset navigation dropdown styles on larger screens
+      @media (min-width: $breakpoint-navigation-threshold) {
```

Comment:
I wonder if we can avoid doing that or make it less repetitive.

Right not it seems that these styles are defined for dropdowns, then overridden by sliding styles, just to be "reset" again to the same original values.

Especially that some of the values we reset are not visibly set here (height, position, transform, width…) - I guess they come from %navigation-slide, but it's not very clear.

I guess it would be good to jump on a call and discuss some not obvious parts.

Reply:
I tried extracting the styles being reset but the nav and dropdown only had about 2 similar styles that could be extracted which means I'd have had to create 2 different mixins for both of them. I don't think all that is necessary so I left a comment stating some of the styles are being reset are from `%navigation-slide`

Reply:
OK, thanks, that's fine.